purepyvmware/lib/datastore_utils.py

```diff
@@ -0,0 +1,175 @@
+from pyVmomi import vim
+
+
+def compare_identifier(device):
+    """ Compare device canonical name to the Pure Storage identifier.
+
+    Args:
+        device (string): Device canonical names. (See below for examples)
+
+    Returns:
+        pure_devices (list): List of confirmed Pure Storage backed devices.
+    """
+    # Examples of valid device canonical name formats:
+    # naa.624a93703b7b308d98f9425e000113e9
+    # eui.003b7b308d98f94224a9375e00018816
+
+    pure_device = None
+    if device.startswith('naa'):
+        if device[5:11] == self.pure_identifier:
```

Comment:
What is self referring to here? It doesn't look like we're in a class right now?

Reply:
Oops, I decided to move some code outside of the class and make it a helper function and didn't remove the reference to "self". I have resolved this.

-jhop